### DISCUSSION TITLE
Labels dont have same vertical size as buttons, inputs, selects...

### DISCUSSION THREAD

**Discussion by batata004** [fomantic/Fomantic-UI#1792] [Elements (button, image, label, loader, ...)]
One thing that I really link on FUI is that you can trust in the size (medium, big, large...) and apply it to buttons, inputs, selects (dropdown)... and they will look identical in vertical size and in many case they share the same padding, margin... However with labels, that is not the case. For example: the `medium` size is the default for buttons, inputs... and their vertical height is the same and if I place a `<div class="ui label">`  at the side of those elements, the vertical size of the label will be much much smaller. Ok, no big deal, I tried applying the `big` and `large` class to the label, however there is no good fit (not even close), the `large` is noticeably smaller and the `big` is noticeacly bigger.

So I cant even find a defined size for the label that matches the same size of a `medium` button, input, dropdown... Or am I doing something wrong?

Also notice that the margin is different, the label is not with equal distance from the other buttons itself.

![image](https://user-images.githubusercontent.com/8988923/102136995-eb32d580-3e38-11eb-92aa-0a680e1188c8.png)

![image](https://user-images.githubusercontent.com/8988923/102137053-ff76d280-3e38-11eb-8254-f6995c0f9755.png)

**Comment by lubber-de** [comment]
That's intention and should not be considered a bug.
Labels are meant to be in flow with a Textline, that's why their equal size name is overall smaller than buttons or inputs, and to visually differ between them.

To get a better idea of why that matters, look at [Horizontal labels](https://fomantic-ui.com/elements/label.html#horizontal), [Pointing labels](https://fomantic-ui.com/elements/label.html#pointing) , [Floating Labels](https://fomantic-ui.com/elements/label.html#floating) or Labels inside Inputs for example  in the Multiple Selection Dropdown

However, there exists adjustmensts already to match the same size/margin/padding when used in [labeled input](https://fomantic-ui.com/elements/input.html#labeled) or [labeled button](https://fomantic-ui.com/elements/button.html#labeled)

Of course you can adjust the values for padding/margin in `label.less` according to your needs, but we should not change the default theme

**Comment by batata004** [comment]
Thank you and I totally agree with you. However, sometimes I just want some label alongside buttons to display some information (like in the print above) and the `labeled button` are not a great idea because they allow user to interact with it (hovering the mouse, changes its color, the default cursor is `pointer`, clicking down the mouse also change the color of the button...) and the user may think that the button is broken if nothing happens when they click the button (and nothing should happen since it's supposed to be a label and not a button).

I know I can override most settings in custom css files but in many projects I import FUI from CDN so I need to add the changes inline to the elements or override styles on a separated stylecheet.

Anyway, if there is no way I can at least style the label to the same size as button (even using a different name) I will stick with a `<div style="height:30px; border-radius:5px; border:1px solid grey; color:grey; padding:7px;">My Label</div>`  - it's ugly but gets the job done :)

**Comment by lubber-de** [comment]
Instead of inline styling (each label) for your usecase, i suggest to override it at a central place , if you want  it for all your labels. From what i see, only the padding for a `large label` needs adjustments, so it will 
look just like a button in size.

```css
.ui.large.label {
 padding: 0.78571429em 1.5em; 
}
```
See https://jsfiddle.net/lubber/bze3r1uL/11/
![image](https://user-images.githubusercontent.com/18379884/102712311-78a27980-42c0-11eb-871a-a8bbb944d6a1.png)

**Comment by batata004** [comment]
Hi my friend, I had indeed realized that I should only change the padding (I even changed it inline, inside the `style` attr because I have only 3 labels on my app) but also the `margin-right` to set it to `3.5px` which is the same of the buttons (the label has `2px` of margin). But I still think the padding value is still wrong, it should be a bit smaller, check print below (I took it from your jsfiddle at https://jsfiddle.net/lubber/bze3r1uL/11/ ). The strange thing is that on my own app the size is perfect with the numbers you gave me, only on your jsfiddle the vertical size looks a little bit off. In my app it's perfect.

 I think that at least the spacing (margin) of labels and buttons should default to the same value since I believe it is pretty common to people use buttons/inputs/dropdowns and labels alongside. I think that even the sizing (medium, small, large...) should be exactly the same to all the components, but I am pretty sure you already know much better use cases that maybe the sizes should not match.

On of the things I loved since the beginning with SUI (and later with FUI) was consistensy: 99% of elements align "just right", no hacks, no fixes... 99% of elements have a really similar aesthetic, spacing, border, light/dark/pastel colors... when you use same sizes or colors. Actually this is the first time I have to deal with a component not behaving as expected (label), but everything else is pretty easy and intuitive!

![image](https://user-images.githubusercontent.com/8988923/102716506-109d6480-42bb-11eb-9b04-21a24d448019.png)

**Comment by ko2in** [comment]
The fiddle from @lubber-de looks OK to me. The label and buttons have the same height (36px) on Microsoft Edge, Chrome. 
![fui-label](https://user-images.githubusercontent.com/930315/102722696-411fe580-4331-11eb-80cb-f5dc4c0985c1.gif)

On my Firefox, the label's height is slightly different (36.5px) but not easy to catch by my eye and totally looks the same. Here's from Firefox.
![fui-label-firefox](https://user-images.githubusercontent.com/930315/102722789-e6d35480-4331-11eb-8741-949be033b39d.png)

**Comment by batata004** [comment]
@ko2in Since you told me on your computer everything looked perfectly, I spent almost 30 minutes investigating why my latest Chrome, on my Windows 10 (with latest update) was showing difference in margin-right and height of the label and the buttons alongside (and in your Chrome you reported everything is ok). I discovered the reason and I have no idea if there is a way to fix it. 

Please, check the first 2 images (the first is the print and the second is the zoom of a particular area of the print) and check the second 2 images.

**FIRST 2 IMAGES**

![image](https://user-images.githubusercontent.com/8988923/102724715-4d378300-42f0-11eb-9fe9-909d9778ccf6.png)
![image](https://user-images.githubusercontent.com/8988923/102724745-a0113a80-42f0-11eb-8950-c2ea4484aebc.png)

**SECOND 2 IMAGES**

![image](https://user-images.githubusercontent.com/8988923/102724777-cc2cbb80-42f0-11eb-8388-3d55624a9533.png)
![image](https://user-images.githubusercontent.com/8988923/102724800-fbdbc380-42f0-11eb-9548-baa53a886a83.png)

By default, my windows laptop came with 125% zoom on display and I use it like that since I bought it and after investigating the "inner working" of my laptop, I tried switching the display zoom to 100% and the height problem disappeared BUT the margin problem still remains. **In both zooms above you can see the spacing between elements is different**.

Anyway, I am not "pedant" and I am not "complaining" (I am not in the position to complain about anything, you guys are doing an awesome job) but I just reported this information back to you if you want to use it for solve something. It would be nice to have an app/website that works pixel perfect accross devices with different zoom situations but I know that maybe this is a "subpixel" problem and every device/browser may implement a different code to make a subpixel bigger or smaller!

If you want, I will close this, ok?

**Reply by ko2in** [reply]
@batata004 The right margin is correct as you mentioned. The label has less margin value (.14285714em) than the button (.25em) by design. But, I don't see any problem with the height as you mentioned with padding issue in @lubber-de fiddle.

Even I zoom my browser to 400%, I don't see significant result like your screenshot. Just let you know that I can't reproduce though. I am testing on my Ubuntu machine. Maybe, I'll check on my Windows 10 PC and see if there's any difference like your screenshots.

**Comment by batata004** [comment]
@ko2in Thanks my friend for your answer. I dont recommend you using the in-browser zoom feature because it does not keep things proportional, the subpixel will certainly happens! The zoom of browsers is not reliable in order to keep things in relative proportion. When I talked to you about the zoom issue I am having, is about the scaling/zoom that Windows 10 itself applies to the whole OS. The strange is that on FF, even with the Windows 10 zoom, everything works the same, but on Chrome it only works when I set 100% scale/zoom in the windows 10 whole system.

Anyway, I dont know if this is easy to fix (or even possible to fix) because it probably varies a lot how each system/browser deals with subpixel when there is zoom applied (wheter in the OS or in the browser, but in the browseer it's expected that the relative sizes of things are not kept exactly right, because of the subpixl problem).

**Reply by ko2in** [reply]
Well. I can't reproduce on my Windows machine either. I changed my display setting to 150% scale and everything look the same.

![display](https://user-images.githubusercontent.com/930315/102793562-7f241480-43d8-11eb-8a7b-7d781eaebb0b.png)

![fui-label](https://user-images.githubusercontent.com/930315/102793715-af6bb300-43d8-11eb-8c31-e562089eb37a.png)

**Comment by batata004** [comment]
Mine I was using 125% do reproduce the problem. Anyway, this is not a big deal :) What would be nice is that buttons/labels share the same sizing/color scheme, but I understand if this is not possible. I am closing this issue, have a nice christmas for all of you!
